### PR TITLE
etcd-runner: election mode

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

**Comment by gyuho** [comment]
LGTM. Defer to @xiang90

File: tools/functional-tester/etcd-runner/main.go [hunk 1]
```diff
@@ -51,6 +53,35 @@ func main() {
...
+		rcs[i].c = randClient(eps)
+		e := concurrency.NewElection(rcs[i].c, "electors")
+		v := fmt.Sprintf("%d", i)
+		rcs[i].acquire = func() error { return e.Campaign(ctx, v) }
+		rcs[i].validate = func() error {
+			if l, err := e.Leader(); err != nil && l != v {
+				return fmt.Errorf("expected leader %q, got %q", l, v)
+			}
+			if cnt++; cnt != 1 {
+				return fmt.Errorf("bad lock; count: %d", cnt)
```
Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: too many leader?

File: tools/functional-tester/etcd-runner/main.go [hunk 2]
```diff
@@ -51,6 +53,35 @@ func main() {
...
+			}
+			return nil
+		}
+		rcs[i].release = func() error {
+			if err := e.Resign(); err != nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: should all client do resign? only leader should do this?

File: tools/functional-tester/etcd-runner/main.go [hunk 3]
```diff
@@ -94,67 +125,94 @@ func runLeaseRenewer(eps []string) {
...
+		m := concurrency.NewMutex(rcs[i].c, "racers")
+		rcs[i].acquire = func() error { return m.Lock(ctx) }
+		rcs[i].validate = func() error {
+			if cnt++; cnt != 1 {
+				return fmt.Errorf("bad lock; count: %d", cnt)
+			}
+			return nil
+		}
+		rcs[i].release = func() error {
+			if err := m.Unlock(); err != nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: should all clients do unlock?

**Comment by xiang90** [comment]
@heyitsanthony What is the current testing result? Running OK? Or has issues?

File: tools/functional-tester/etcd-runner/main.go (hunk 3 above)
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: it should be the same behavior as before

File: tools/functional-tester/etcd-runner/main.go (hunk 2 above)
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: it's a nop for followers

File: tools/functional-tester/etcd-runner/main.go (hunk 1 above)
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: I reworked this

File: tools/functional-tester/etcd-runner/main.go (hunk 3 above)
Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: i misread. sorry. this is correct.

**Comment by heyitsanthony** [comment]
@xiang90 it works but I haven't tried fault injection yet

**Comment by xiang90** [comment]
lgtm

**Pull request merged**